app/src/common/shared/com/igalia/wolvic/ui/widgets/WindowWidget.java

```diff
@@ -561,6 +561,7 @@ private void hidePanel(boolean switchSurface) {
             unsetView(mLibrary, switchSurface);
             mLibrary.onHide();
             mViewModel.setCurrentContentType(Windows.ContentType.WEB_CONTENT);
+            mViewModel.setUrl(mSession.getCurrentUri());
```

Comment:
I am not totally sure we need this. The URL seems to be correctly stored already. It's more like the UI is not updated. For example if you follow the steps of the description and at the end you move the window or recenter the view then the correct URL is shown

Reply:
Note that the method `showPanel()` updates the URL field in the view model:

        mViewModel.setCurrentContentType(contentType);
        mViewModel.setUrl(contentType.URL);

The additional line in this PR reverts that change.